beacon-chain/sync/subscriber_beacon_blocks.go

```diff
@@ -55,6 +58,78 @@ func (s *Service) beaconBlockSubscriber(ctx context.Context, msg proto.Message)
 	return err
 }
 
+// reconstructAndBroadcastBlobs processes and broadcasts blob sidecars for a given beacon block.
+// This function reconstructs the blob sidecars from the EL using the block's KZG commitments,
+// broadcasts the reconstructed blobs over P2P, and saves them into the blob storage.
+func (s *Service) reconstructAndBroadcastBlobs(ctx context.Context, block interfaces.ReadOnlySignedBeaconBlock) {
+	startTime, err := slots.ToTime(uint64(s.cfg.chain.GenesisTime().Unix()), block.Block().Slot())
+	if err != nil {
+		log.WithError(err).Error("Failed to convert slot to time")
+	}
+
+	blockRoot, err := block.Block().HashTreeRoot()
+	if err != nil {
+		log.WithError(err).Error("Failed to calculate block root")
+		return
+	}
+
+	if s.cfg.blobStorage == nil {
+		return
+	}
+	indices, err := s.cfg.blobStorage.Indices(blockRoot)
+	if err != nil {
+		log.WithError(err).Error("Failed to retrieve indices for block")
+		return
+	}
+	for _, index := range indices {
+		if index {
+			blobExistedInDBCount.Inc()
+		}
+	}
+
+	// Reconstruct blob sidecars from the EL
+	blobSidecars, err := s.cfg.executionReconstructor.ReconstructBlobSidecars(ctx, block, blockRoot, indices)
+	if err != nil {
+		log.WithError(err).Error("Failed to reconstruct blob sidecars")
+		return
+	}
+	if len(blobSidecars) == 0 {
+		return
+	}
+
+	// Refresh indices as new blobs may have been added to the db
+	indices, err = s.cfg.blobStorage.Indices(blockRoot)
```

Comment:
this way of calling it twice sort of bothers me but don't have a better solution to suggest